R/intervals_support_funs.R

```diff
@@ -0,0 +1,363 @@
+#'   cmax = c(TRUE, TRUE, TRUE),
+#'   impute = c("start_conc0,start_predose", "start_predose", "start_conc0"),
+#'   analyte = c("Analyte1", "Analyte2", "Analyte1")
+#' )
+#'
+#' o_data <- PKNCAdata(o_conc, o_dose, intervals = intervals)
+#'
+#' # Apply interval_remove_impute function
+#' o_data <- interval_remove_impute(data = o_data,
+#'                                  target_impute = "start_conc0",
+#'                                  target_params = "half.life",
+#'                                  target_groups = data.frame(analyte = "Analyte1"))
+#' @export
+interval_remove_impute <- function(data, ...) {
+  UseMethod("interval_remove_impute", data)
+}
+
+#' @export
+interval_remove_impute.PKNCAdata <- function(data, target_impute, target_params = NULL,
+                                             target_groups = NULL, ...) {
+  # If the impute column is not present in the intervals...
+  if (!"impute" %in% colnames(data$intervals)) {
+    if (is.null(data$impute) || is.na(data$impute)) {
+      # a. If neither is in the global impute method, return the data as it is
+      warning("No default impute column or global method identified. No impute methods to remove")
+      return(data)
+    }
+
+    # b & c. If global impute exists
+    if (isTRUE(!is.na(data$impute))) {
+      if (is.null(target_params) && is.null(target_groups)) {
+        # b. If it is in the global impute and no target parameters or groups, remove global impute
+        data$impute <- remove_impute_method(data$impute, target_impute)
+        return(data)
+      } else {
+        # c. If it is in the global impute but target parameters or groups, creates an impute column
+        data$intervals$impute <- data$impute
+        data$impute <- NA_character_
+      }
+    }
+  }
+
+  data$intervals <- interval_remove_impute.data.frame(data$intervals, target_impute,
+                                                      target_params, target_groups)
+  data
+}
+
+
+#' Remove impute method from the impute column
+#'
+#' This is an internal helper function used to remove an impute method from the impute column.
+#'
+#' @param impute_vals Character vector of impute methods.
+#' @param target_impute The imputation method to be removed.
+#' @return A character string or vector without the specified impute method.
+#' @details Resulting empty string values are replaced with NA_character_.
+#' @keywords internal
+remove_impute_method <- function(impute_vals, target_impute) {
+  # Make sure the character vector has length
+  if (length(impute_vals) == 0) return(impute_vals)
+
+  # Remove the impute from the other methods in each value
+  impute_vals <- ifelse(is.na(impute_vals), "", impute_vals)
+  impute_vals <- strsplit(impute_vals, split = "[ ,]+") |>
+    lapply(FUN = setdiff, target_impute) |>
+    vapply(FUN = paste, collapse = ",", FUN.VALUE = "")
+
+  # Replace empty strings with NA_character_
+  ifelse(impute_vals == "", NA_character_, impute_vals)
+}
+
+#' @export
```

Comment:
issue: No need to export this function, I would use `@keywords internal` instead.
```suggestion
#' @keywords internal
```